tasks/connectors/_sample_task/sample_task.rb

```diff
@@ -0,0 +1,174 @@
+              required: false,
+              default: "output/sample",
+              description: "If set, will write a file upon completion. Path is relative to #{$basedir}" }
+          ]
+        }
+      end
+
+      def run(opts)
+        super
+
+        initialize_options
+
+        client = Kenna::Toolkit::Sample::Client.new(@host, @api_token)
+
+        @schedule_ids.each do |schedule_id|
+          last_schedule_scan = client.get_last_schedule_scan(schedule_id)
+          if last_schedule_scan
+            total_issues = @issue_severities.sum { |key| (last_schedule_scan["issue_counts"][key]["total"] || 0).to_i }
+            scan_id = last_schedule_scan["id"]
+            print_good("Found scan ##{scan_id} for schedule ##{schedule_id} with #{total_issues} issues with severities #{@issue_severities}.")
+            pos = 0
+
+            while pos < total_issues
+              last_scan = client.get_scan(scan_id, @issue_severities, pos, @batch_size)
+              issues = last_scan["issues"]
+              issues.each do |issue|
+                asset = extract_asset(issue)
+                finding = extract_finding(issue)
+                definition = extract_definition(issue)
+
+                create_kdi_asset_finding(asset, finding)
+                create_kdi_vuln_def(definition)
+              end
+
+              print_good("Processed #{[pos + @batch_size, total_issues].min} of #{total_issues} issues for scan ##{scan_id}.")
+              kdi_upload(@output_directory, "sample_scan_#{scan_id}_report_#{pos}.json", @kenna_connector_id, @kenna_api_host, @kenna_api_key, @skip_autoclose, @retries, @kdi_version)
+              pos += @batch_size
+            end
+          else
+            print("No scan found for schedule #{schedule_id}")
+          end
+        end
+        kdi_connector_kickoff(@kenna_connector_id, @kenna_api_host, @kenna_api_key)
+      rescue Kenna::Toolkit::Sample::Client::ApiError => e
+        fail_task e.message
+      end
+
+      private
+
+      def initialize_options
+        @host = @options[:sample_api_host]
+        @schedule_ids = extract_list(:sample_schedule_id)
+        @issue_severities = extract_list(:sample_issue_severity, %w[info low medium high])
+        @api_token = @options[:sample_api_token]
+        @output_directory = @options[:output_directory]
+        @kenna_api_host = @options[:kenna_api_host]
+        @kenna_api_key = @options[:kenna_api_key]
+        @kenna_connector_id = @options[:kenna_connector_id]
+        @batch_size = @options[:batch_size].to_i
+        @skip_autoclose = false
+        @retries = 3
+        @kdi_version = 2
+      end
+
+      def extract_list(key, default = nil)
+        list = (@options[key] || "").split(",").map(&:strip)
+        list.empty? ? default : list
+      end
+
+      SEVERITY_VALUE = {
+        "info" => 0,
+        "low" => 3,
+        "medium" => 6,
+        "high" => 10
+      }.freeze
+
+      def extract_asset(issue)
```

Comment:
@jscalvo  i think there was benefit in the old sample to having all the possible values with certain things commented out. either that or there we should have 2 with one having all kdi fields and this one being a specific "findings" sample.

Reply:
Ok, I Agree, let me correct this.